core/primitives/transaction_validity.hpp

```diff
@@ -53,123 +54,139 @@ namespace kagome::primitives {
      */
     TransactionLongevity longevity;
 
-    bool operator==(const Valid &rhs) const {
+    /**
+     * @brief A flag indicating if the transaction should be propagated to other
+     * peers. By setting `false` here the transaction will still be considered
+     * for including in blocks that are authored on the current node, but will
+     * never be sent to other peers.
+     */
+    bool propagate;
+
+    bool operator==(const ValidTransaction &rhs) const {
       return priority == rhs.priority and requires == rhs.requires
-             and provides == rhs.provides and longevity == rhs.longevity;
+             and provides == rhs.provides and longevity == rhs.longevity
+             and propagate == rhs.propagate;
     }
 
-    bool operator!=(const Valid &rhs) const {
+    bool operator!=(const ValidTransaction &rhs) const {
       return !operator==(rhs);
     }
   };
 
   /// Transaction is invalid. Details are described by the error code.
-  struct Invalid {
-    uint8_t error;
-
-    bool operator==(const Invalid &rhs) const {
-      return error == rhs.error;
-    }
-    bool operator!=(const Invalid &rhs) const {
-      return error != rhs.error;
-    }
-  };
-
-  /// Transaction validity can't be determined.
-  struct Unknown {
-    uint8_t error;
-
-    bool operator==(const Unknown &rhs) const {
-      return error == rhs.error;
-    }
-    bool operator!=(const Unknown &rhs) const {
-      return error != rhs.error;
-    }
+  enum class InvalidTransaction : uint8_t {
```

Comment:
Values should start with 1 and also naming is unconventional.

Reply:
This is not common outcome result error. It should be encoded/decoded the same way as in rust, therefore ordering of items should be the same and they should start from 0 as in rust
https://github.com/paritytech/substrate/blob/a31c01b398d958ccf0a24d8c1c11fb073df66212/core/sr-primitives/src/transaction_validity.rs#L36